docs/config.md

```diff
@@ -86,3 +89,20 @@ further settings are passed on via the following:
 
     merge-tools.kdiff3.program = "kdiff3"
     merge-tools.kdiff3.edit-args = ["--merge", "--cs", "CreateBakFiles=0"]
+
+# Alternative ways to specify configuration settings
+
+The location of the `jj` config file can be overriden with the `JJ_CONFIG` environment variable. If it is not empty, it should contain the path to a TOML file that will be used instead of any configuration file in the default locations. For example,
+
+    env JJ_CONFIG=/dev/null jj log       # Ignores any settings specified in the config file.
+
+You can use one or more `--config` options on the command line to specify additional configuration settings. This overrides settings defined anywhere else. For example,
+
+    jj --config='ui.color="always"' --config='ui.difftool="kdiff3"' split
+
+Config specified this way must be valid TOML. In paritcular, string values must be surrounded by quotes. To pass these quotes to `jj`, most shells require surrounding those quotes with single quotes as shown above.
```

Comment:
What if we called this flag `--config_toml` and, later, introduced a `--config` flag with a simpler syntax for simple configuration. It would be designed so that you could say `--config ui.color=always`. The logic would be something like:

-- Split the string on the regex `\s*=\s*`. The left side is the key name, the right side is the value.
-- The value is interpreted as a boolean if it's `true` or `false`, otherwise it's a string.
-- The key is split on the dots to get the nesting.

There would be no way to set a key like `merge-tools.meld.diff-args` with `--config`, or to set more than one key per `--config`, but you could use `--config_toml` for complete generality.

WDYT?

Reply:
I like the proposal :)

Reply:
Sounds good :) I changed the name and left a TODO.